v2/tools/generator/internal/codegen/pipeline/add_cross_resource_references.go

```diff
@@ -165,22 +166,41 @@ func DoesPropertyLookLikeARMReference(prop *astmodel.PropertyDefinition) bool {
 }
 
 func makeResourceReferenceProperty(idFactory astmodel.IdentifierFactory, existing *astmodel.PropertyDefinition) *astmodel.PropertyDefinition {
+	_, isSlice := astmodel.AsArrayType(existing.PropertyType())
+	_, isMap := astmodel.AsMapType(existing.PropertyType())
+	propertyNameSuffix := "Reference"
+	if isSlice || isMap {
+		propertyNameSuffix = "References"
+	}
```

Comment:
Should we check for both? An array of references with a singular suffix (or a single reference with a plural suffix) might just be poorly named (and maybe named that way for backward compatibility).

Reply:
Not sure I follow you. I am checking for both already here?

Do you mean swapping the hardcoded `Id` -> `Reference` replacement with `Id` -> `propertyNameSuffix`? I've done that just now

Reply:
Apologies, I thought this was the suffix you were looking for. 😊